### PR TITLE
Refactor Digital Signature - LRQA-82436 | master

### PR DESCRIPTION
**Pull request by juliosilveira77**
https://liferay.atlassian.net/browse/LRQA-82436

### CODE REVIEW

**Comment by juliosilveira77** [comment]
ci:test:sf

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 1]
```diff
@@ -125,166 +119,96 @@ definition {
...
 
-		LexiconEntry.gotoAdd();
+			LexiconEntry.gotoAdd();
+		}
 
-		AssertElementPresent(
-			fieldLabel = "Envelope Name",
-			locator1 = "FormViewBuilder#FIELD_LABEL_REQUIRED");
+		task ("Then it must have the Recipient Full Name field") {
+			DigitalSignature.digitalSignatureField(fieldList = "Email Subject,Envelope Name,Recipient Full Name,Recipient Email");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:178]: Could you change this macro name to : `canViewRequiredFields`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 2]
```diff
@@ -125,166 +119,96 @@ definition {
...
+				portlet = "Digital Signature");
 
-		LexiconEntry.gotoAdd();
+			LexiconEntry.gotoAdd();
+		}
 
-		AssertElementPresent(
-			fieldLabel = "Envelope Name",
-			locator1 = "FormViewBuilder#FIELD_LABEL_REQUIRED");
+		task ("Then it must have the Recipient Full Name field") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:177]: Please change this macro to: `"Then All required fields are being displayed"`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 3]
```diff
@@ -125,166 +119,96 @@ definition {
...
+				modalLabel = "File Extensions Not Supported");
 
-			Click(locator1 = "Button#CONTINUE");
-
-			SelectFrameTop();
-
-			AssertElementNotPresent(
-				fileTitle = "Document_3",
-				locator1 = "DigitalSignatureEnvelopeSettings#DIGITAL_SIGNATURE_ENVELOPE_ADDED_FILE");
+			DigitalSignature.fileNotDigitalSignature(fileName = "Document_3");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:132]: Please change this macro name to: `canNotSeeUnsupportedFiles`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 4]
```diff
@@ -125,166 +119,96 @@ definition {
...
-			SelectFrame(locator1 = "Modal#MODAL");
-
-			AssertElementPresent(
-				locator1 = "DigitalSignatureEnvelopeSettings#DIGITAL_SIGNATURE_MODAL_TITLE",
-				modalTitle = "File Extensions Not Supported");
-
-			AssertElementPresent(
-				fileTitle = "Document_3",
-				locator1 = "DigitalSignatureEnvelopeSettings#DIGITAL_SIGNATURE_MODAL_UNSUPPORTED_FILE");
+			DigitalSignature.extensionsNotSupported(
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:128]: Please change this macro name to:` viewNotSupportedExtensionsError`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 5]
```diff
@@ -34,89 +36,81 @@ definition {
...
+		}
 
-		Click(locator1 = "Button#REMOVE_IMAGE_LAST");
+		task ("When user test Add Recipient") {
+			DigitalSignature.addRecipient();
+		}
 
-		AssertElementPresent(locator1 = "Button#TRASH_DISABLED");
+		task ("Then user test remove Recipient") {
+			DigitalSignature.removeRecipient();
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:106]: Please change this task to : `Then the user remove Recipient`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 6]
```diff
@@ -34,89 +36,81 @@ definition {
...
 
-		AssertClick(
-			key_text = "Add Recipient",
-			locator1 = "Button#ANY_WITH_SPAN",
-			value1 = "Add Recipient");
+			LexiconEntry.gotoAdd();
+		}
 
-		Click(locator1 = "Button#REMOVE_IMAGE_LAST");
+		task ("When user test Add Recipient") {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:101]: Please change this task to : `When the user Add  Recipient`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 7]
```diff
@@ -34,89 +36,81 @@ definition {
...
 	test CanDeleteRecipients {
-		DigitalSignature.enableDigitalSignature(siteStrategy = "Always Inherit");
-
-		Navigator.openURL();
-
-		ProductMenu.gotoPortlet(
-			category = "Content & Data",
-			portlet = "Digital Signature");
+		task ("Given that user Test goes to digital signature") {
+			Navigator.openURL();
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:92]: Please change this task to : `Given the user goes to digital signature`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 8]
```diff
@@ -34,89 +36,81 @@ definition {
...
-				value1 = "Add Recipient");
+		task ("When user test Add 10 Recipients") {
+			DigitalSignature.addRecipient(addList = "1,2,3,4,5,6,7,8,9");
 		}
 
-		AssertElementNotPresent(
-			key_text = "Add Recipient",
-			locator1 = "Button#ANY_WITH_SPAN");
+		task ("Then user test is unable to add more Recipient") {
+			DigitalSignature.addRecipientNotPresent(notPresent = "Add Recipient");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:84]: Please change this macro name to: `canNotViewAddRecipient`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase [hunk 9]
```diff
@@ -10,17 +10,19 @@ definition {
...
+		//		category = "Configuration",
+		//		panel = "Control Panel",
+		//		portlet = "Instance Settings");
+		//	SystemSettings.gotoConfiguration(
+		//		configurationCategory = "Digital Signature",
+		//		configurationName = "Digital Signature Configuration",
+		//		configurationScope = "Virtual Instance Scope");
+		//	FormFields.enableCheckbox(fieldName = "enabled");
+		//	DigitalSignature.enableDigitalSignature(siteStrategy = "Always Inherit");
+		//}
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:24]: Please remove this comments

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 10]
```diff
@@ -228,10 +277,24 @@ definition {
...
+		AssertElementPresent(locator1 = "Button#TRASH_DISABLED");
+	}
+
 	macro selectDMFolder {
 		Click.javaScriptClick(
 			key_fieldName = ${checkboxDM},
 			locator1 = "Checkbox#CHECKBOX_NAME");
 	}
 
+	macro unsupportedFile {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:292]: Please change this macro name to: `verifyUnsupportedFileError`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 11]
```diff
@@ -205,6 +232,28 @@ definition {
...
+		AssertElementPresent(
+			locator1 = "DigitalSignatureEnvelopeSettings#DIGITAL_SIGNATURE_MODAL_TITLE",
+			modalTitle = ${modalLabel});
+
+		AssertElementPresent(
+			fileTitle = ${documentTitle},
+			locator1 = "DigitalSignatureEnvelopeSettings#DIGITAL_SIGNATURE_MODAL_UNSUPPORTED_FILE");
+	}
+
+	macro fileNotDigitalSignature {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:247]: Please change this macro name to: `canNotSeeUnsupportedFiles`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 12]
```diff
@@ -205,6 +232,28 @@ definition {
 		SystemSettings.saveConfiguration();
 	}
 
+	macro extensionsNotSupported {
+		SelectFrame(locator1 = "Modal#MODAL");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:236]: Please change this macro name to: `viewNotSupportedExtensionsError`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 13]
```diff
@@ -161,6 +180,14 @@ definition {
 		Navigator.openSpecificURL(url = ${value});
 	}
 
+	macro digitalSignatureField {
+		for (var fieldName : list ${fieldList}) {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:184]: Could you change this macro name to : `canViewRequiredFields`

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 14]
```diff
@@ -1,5 +1,20 @@
...
+				key_text = "Add Recipient",
+				locator1 = "Button#ANY_WITH_SPAN",
+				value1 = "Add Recipient");
+		}
+	}
+
+	macro addRecipientNotPresent {
+		AssertElementNotPresent(
+			key_text = ${notPresent},
+			locator1 = "Button#ANY_WITH_SPAN");
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:15]: Could you remove this variant: `${not present}` and put the same value you out above: `"Add Recipient"` so when you call this macro you won't need call the this variant.

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro [hunk 15]
```diff
@@ -1,5 +1,20 @@
...
+	macro addRecipient {
+		for (var list : list ${addList}) {
+			AssertClick(
+				key_text = "Add Recipient",
+				locator1 = "Button#ANY_WITH_SPAN",
+				value1 = "Add Recipient");
+		}
+	}
+
+	macro addRecipientNotPresent {
```
Comment (MGabriellaS) [inline modules/apps/digital-signature/digital-signature-test/src/testFunctional/macros/DigitalSignature.macro:12]: Could you change the name to `canNotViewAddRecipient`?

**Review by MGabriellaS — COMMENTED** [review]
@juliosilveira77 Could you take a look when you get a chance?

File: modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase (hunk 2 above)
Comment (juliosilveira77) [reply modules/apps/digital-signature/digital-signature-test/src/testFunctional/tests/DigitalSignatureEnvelopeSettings.testcase:177]: Task

**Comment by juliosilveira77** [comment]
ci:test:sf

**Comment by MGabriellaS** [comment]
ci:forward

**Pull request closed without merging**